File: 백준/Silver/13305. 주유소/주유소.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int N = Integer.parseInt(br.readLine());
        long[] distance = new long[N - 1];
        long[] cost = new long[N];

        StringTokenizer st = new StringTokenizer(br.readLine(), " ");
        for (int i = 0; i < N-1; i++) {
            distance[i] = Long.parseLong(st.nextToken());
        }

        st = new StringTokenizer(br.readLine(), " ");
        for (int i = 0; i < N; i++) {
            cost[i] = Long.parseLong(st.nextToken());
        }

        long tmp = cost[0];
        long sum = 0;

        for (int i = 0; i < N - 1; i++) {
            if(cost[i] < tmp) {
                tmp = cost[i];
            }
            sum = sum + (tmp * distance[i]);
        }
        System.out.println(sum);
    }
}

File: 백준/Silver/13305. 주유소/주유소.py
import sys

N = int(sys.stdin.readline())

road_dis = list(map(int, sys.stdin.readline().split()))
gas_price = list(map(int, sys.stdin.readline().split()))

sum = 0
tmp = gas_price[0]

for i in range(len(road_dis)):
    if tmp > gas_price[i]:
       tmp = gas_price[i]
    sum += tmp * road_dis[i]
 
print(sum)   